index.js

```diff
@@ -176,16 +176,23 @@ Shopify.prototype.graphql = function graphql(data) {
 
   path += '/graphql.json';
 
+  const json = typeof variables !== 'undefined' && variables !== null;
+  const body = json ? JSON.stringify({
+    query: data,
+    variables
+  }) : data;
+
   const url = assign({ path }, this.baseUrl);
   const options = assign({
     timeout: this.options.timeout,
     retries: 0,
     method: 'POST',
-    body: data
+    responseType: 'json',
+    body
   }, url);
 
   options.headers['User-Agent'] = `${pkg.name}/${pkg.version}`;
-  options.headers['Content-Type'] = 'application/graphql';
+  options.headers['Content-Type'] = json ? 'application/json' : 'application/graphql';
```

Comment:
Set this to `'application/graphql'` only when `json` is `false`. The `json` option will take care of adding `'application/json'` when it is set.

Reply:
No longer valid